### PR TITLE
e2e test for get_object_with_options, testing out some data options and when object does not exist

### PR DESCRIPTION
**Pull request by wlmyng**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

Not a new feature, but ci should still pass.
---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-indexer/tests/integration_tests.rs [hunk 1]
```diff
@@ -2,8 +2,9 @@
 // SPDX-License-Identifier: Apache-2.0
 
 // integration test with standalone postgresql database
-#[cfg(feature = "pg_integration")]
+// #[cfg(feature = "pg_integration")]
```
Comment (wlmyng) [inline crates/sui-indexer/tests/integration_tests.rs:5]: will uncomment

File: crates/sui-indexer/tests/integration_tests.rs [hunk 2]
```diff
@@ -387,6 +430,156 @@ pub mod pg_integration_test {
...
+                source_object_id,
+                Some(show_some_content.clone().with_display()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_bcs()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_previous_transaction()),
```
Comment (wlmyng) [inline crates/sui-indexer/tests/integration_tests.rs:513]: happy to take any thoughts you have on rewriting this neater

File: crates/sui-indexer/tests/integration_tests.rs [hunk 3]
```diff
@@ -387,6 +430,156 @@ pub mod pg_integration_test {
...
+        ];
+
+        for (received, expected) in results.iter().zip(expected_results.iter()) {
+            let data = received.object()?;
+            assert_eq!(data, expected);
+            assert_ne!(data.version, initial_full_obj_data.version);
+            assert_eq!(data.object_id, initial_full_obj_data.object_id);
+        }
+
+        // deleted(SuiObjectRef);
```
Comment (wlmyng) [inline crates/sui-indexer/tests/integration_tests.rs:571]: pending figuring out how to delete an object https://mysten-labs.slack.com/archives/C04HS54LHUM/p1679353365380589

File: crates/sui-indexer/tests/integration_tests.rs (hunk 2 above)
Comment (gegaowp) [reply crates/sui-indexer/tests/integration_tests.rs:513]: it is a bit verbose here, but it's okay

File: crates/sui-indexer/tests/integration_tests.rs (hunk 3 above)
Comment (gegaowp) [reply crates/sui-indexer/tests/integration_tests.rs:571]: per Patrick's comment, we can merge coin to delete one.

**Review by gegaowp — APPROVED** [review]
looks good overall.

File: crates/sui-indexer/tests/integration_tests.rs (hunk 3 above)
Comment (wlmyng) [reply crates/sui-indexer/tests/integration_tests.rs]: hehe thanks all

**Pull request merged**